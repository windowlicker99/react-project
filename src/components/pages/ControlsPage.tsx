import React, { FC, useState } from 'react';
import { Tab, TabPanel, TabList, Tabs } from 'react-tabs';
import { useSelector } from 'react-redux';
import { PageSearchbar } from '@components/page-contents/PageSearchbar';
import { Table } from '@components/page-contents/Table';
import { Footer } from '@components/modules/Footer';
import classNames from 'classnames';
import { Preloader } from '@components/common/Preloader';
import { Btn } from '@components/common/form-controls/Btn';
import { selectColumnsWithDropdownList } from '@/store/selectors/columns';
import { controlsTabs } from '@/constants/tabsLists';

export const ControlsPage: FC = () => {
  const [searchText, setSearchText] = useState<string>('');

  const [loading, setLoading] = useState(true);

  const customizedColumns = useSelector(selectColumnsWithDropdownList);

  const [selectedFindingsTabIndex, setSelectedFindingsTabIndex] = useState(0);

  return (
    <Tabs
      className="content"
      id="controls"
      onSelect={setSelectedFindingsTabIndex}
      selectedIndex={selectedFindingsTabIndex}
      selectedTabPanelClassName="box-table box-not-radius"
    >
      <div className="box page-controls">
        <PageSearchbar
          searchText={searchText}
          onSearch={setSearchText}
          onCustomizeColumnsClick={(): null => null} // temp
          btnText="Create Control"
          btnIcon="icon-plus"
          showFilterBtn={false}
        />
      </div>

      <TabList className="tabs-box">
        {controlsTabs.map((tab, i) => (
          <Tab key={tab.name}>
            <Btn
              btnStatus={classNames('btn-stretch', selectedFindingsTabIndex === i ? 'btn-lg-primary' : 'btn-lg-info')}
              btnText={tab.name}
            />
          </Tab>
        ))}
      </TabList>

      {controlsTabs.map((tab) => (
        <TabPanel key={tab.name}>
          <Table
            changeLoadingState={setLoading}
            columns={customizedColumns}
            setSearchText={setSearchText}
            searchText={searchText}
            tableType={tab.type}
            withoutPagination
          />
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
      <Footer />
    </Tabs>
  );
};
