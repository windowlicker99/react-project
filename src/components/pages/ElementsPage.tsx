import React, { FC, useState } from 'react';
import { Tab, TabPanel, TabList, Tabs } from 'react-tabs';
import { useDispatch, useSelector } from 'react-redux';
import { useFormik } from 'formik';
import { PageFilter } from '@components/page-contents/PageFilter';
import { PageSearchbar } from '@components/page-contents/PageSearchbar';
import { Table } from '@components/page-contents/Table';
import { elementsTabsList } from '@constants/tabsLists';
import { ModalCenter } from '@components/modules/ModalCenter';
import { CustomiseModalContent } from '@components/modal-content/CustomiseModalContent';
import { selectColumnsInitialValues, selectColumnsList, selectColumnsFieldsList } from '@store/selectors/columns';
import { updateDisplayedColumns } from '@store/slices/columnsSlice';
import { Preloader } from '@components/common/Preloader';
import { Footer } from '@components/modules/Footer';

export const ElementsPage: FC = () => {
  const dispatch = useDispatch();
  const [searchText, setSearchText] = useState<string>('');
  const [loading, setLoading] = useState(true);

  const [isOpenCustomizeModal, setIsOpenCustomizeColumnsModal] = useState<boolean>(false);
  const initialValues = useSelector(selectColumnsInitialValues);
  const columnsList = useSelector(selectColumnsList);
  const columnsFields = useSelector(selectColumnsFieldsList);

  const formik = useFormik({
    initialValues,
    enableReinitialize: true,
    onSubmit: (values) => {
      dispatch(updateDisplayedColumns({ displayColumnsList: values }));
    },
  });

  const onOpenCustomizeModal = () => {
    setIsOpenCustomizeColumnsModal(true);
  };

  const onApplyCustomizeModal = () => {
    formik.submitForm();
    setIsOpenCustomizeColumnsModal(false);
  };

  const onCloseCustomizeModal = () => {
    formik.resetForm();
    setIsOpenCustomizeColumnsModal(false);
  };

  return (
    <>
      <Tabs className="content" id="elements" selectedTabPanelClassName="box-table elements-content">
        <div className="box page-controls">
          <PageSearchbar
            searchText={searchText}
            onSearch={setSearchText}
            onCustomizeColumnsClick={onOpenCustomizeModal}
            btnText="Customize Columns"
            btnIcon="icon-columns"
          />
          <PageFilter />
          <TabList className="page-tabs">
            {elementsTabsList.map((tab) => (
              <Tab className="page-tabs-item" selectedClassName="page-tabs-item active" key={tab.name}>
                {tab.name}
              </Tab>
            ))}
          </TabList>
        </div>

        {elementsTabsList.map((tab) => (
          <TabPanel key={tab.name}>
            <Table
              columns={columnsList}
              changeLoadingState={setLoading}
              setSearchText={setSearchText}
              searchText={searchText}
              tableType={tab.type}
              showHeader
            />
          </TabPanel>
        ))}
        <Footer />
      </Tabs>

      {loading ? <Preloader /> : null}

      <ModalCenter
        title="Customize columns"
        isOpen={isOpenCustomizeModal}
        onApply={onApplyCustomizeModal}
        onClose={onCloseCustomizeModal}
      >
        <CustomiseModalContent columns={columnsFields} {...formik} />
      </ModalCenter>
    </>
  );
};
