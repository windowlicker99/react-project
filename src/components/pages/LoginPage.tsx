import React, { FC } from 'react';
import { LoginForm } from '@components/page-contents/LoginForm';
import { Logo } from '@components/common/Logo';

export const LoginPage: FC = () => (
  <div className="page-container is-open">
    <div className="login-image">
      <Logo />
    </div>
    <LoginForm />
  </div>
);
