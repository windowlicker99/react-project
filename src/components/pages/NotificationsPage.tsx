import React, { FC } from 'react';
import { PageControls } from '@components/page-contents/PageControls';
import { NotificationsContent } from '@components/page-contents/NotificationsContent';

export const NotificationsPage: FC = () => (
  <div className="content" id="notifications">
    <PageControls />
    <NotificationsContent />
  </div>
);
