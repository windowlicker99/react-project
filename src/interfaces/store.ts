import { ETableTypes, IColumn, ICurrentElement, IElementsTabsColumnsList, IUser } from '@interfaces/interfaces';

export interface IAuth {
  user: IUser;
}
export interface IPagination {
  nextPage: () => void;
  prevPage: () => void;
  pickPage: (indexPage: number) => void;
  disablePrev: boolean;
  disableNext: boolean;
  currentPage: number;
  pageAmount: number;
  currentTab: string;
  totalElement: number;
  totalElementOnPage: number;
  showedElements: number;
}

export interface IVehicleStore {
  currentVehicle: string;
}

export interface IHeaderStore {
  subtitles: string[];
}

export interface IChangeTypeAction {
  type: ETableTypes;
}

export interface IUpdateDisplayedColumnsAction {
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnState {
  type: ETableTypes;
  columns: IElementsTabsColumnsList;
}

export interface IStoreState {
  auth: IAuth;
  pagination: IPagination;
  vehicles: IVehicleStore;
  customizeColumns: ICustomizeColumnState;
  header: IHeaderStore;
  element: IElementStore;
}

export interface IDisplayColumnsList {
  [key: string]: boolean;
}

export interface ICustomizeColumnAction {
  type: ETableTypes;
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnsState {
  [key: string]: IColumn[];
}

export interface IElementStore {
  currentElement: ICurrentElement;
}
