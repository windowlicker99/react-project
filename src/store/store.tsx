import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import { authReducer } from '@store/slices/authSlice';
import { columnsReducer } from '@store/slices/columnsSlice';
import { paginationReducer } from '@store/slices/paginationSlice';
import { vehiclesReducer } from '@store/slices/vehicleSlice';
import { headerReducer } from '@store/slices/headerSlice';
import { elementReducer } from '@store/slices/elementSlice';

const reducers = combineReducers({
  auth: authReducer,
  vehicles: vehiclesReducer,
  customizeColumns: columnsReducer,
  pagination: paginationReducer,
  header: headerReducer,
  element: elementReducer,
});

const persistConfig = {
  key: 'root',
  storage,
  whiteList: ['auth', 'customizeColumns'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: false,
    }),
  ],
});

export const persistor = persistStore(store);
