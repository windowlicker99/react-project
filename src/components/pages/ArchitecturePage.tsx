import React, { useState, FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ArchitectureControls } from '@components/page-contents/ArchitectureControls';
import { Btn } from '@components/common/form-controls/Btn';
import { ArchitectureScheme } from '@components/page-contents/ArchitectureScheme';
import { ModalCenter } from '@components/modules/ModalCenter';
import { AssignRA } from '@components/page-contents/AssignRA';
import { ArchitectureAside } from '@components/common/ArchitectureAside';
import { ArchitectureDetailsContent } from '@components/common/aside-content/ArchitectureDetailsContent';
import { EAside } from '@interfaces/enums';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { getCurrentVehicle } from '@utils/getCurrentVehicle';
import { IGatewayResponse, IVehicle } from '@interfaces/interfaces';
import { DomainDetails } from '@components/page-contents/DomainDetails';
import { ECUDetails } from '@components/page-contents/ECUDetails';
import { getVehicleGateway } from '@utils/getVehicleGateway';
import { setCurrentElement } from '@store/slices/elementSlice';
import { InitiateRiskAssesment } from '@components/page-contents/InitiateRiskAssesment';
import { currentElementSelector } from '@store/selectors/element';
import { getSchemeData } from '@utils/getSchemeData';
import { IArchitectureDomain } from '@interfaces/ArchitectureScheme/interfaces';
import { Preloader } from '@components/common/Preloader';
import { setHeader } from '@/store/slices/headerSlice';

export const ArchitecturePage: FC = () => {
  const dispatch = useDispatch();
  const [vehicle, setVehicle] = useState<IVehicle>(null);
  const [schemeData, setSchemeData] = useState<IArchitectureDomain[]>(null);
  const [isOpenDomainModal, setIsOpenDomainModal] = useState<boolean>(false);
  const [isOpenECUModal, setIsOpenECUModal] = useState<boolean>(false);
  const [isOpenHistoryView, setIsOpenHistoryView] = useState<boolean>(false);
  const [isOpenRAModal, setIsOpenRAModal] = useState<boolean>(false);
  const [isOpenInitiateView, setIsOpenInitiateView] = useState<boolean>(false);
  const [gateway, setGateway] = useState<IGatewayResponse>(null);
  const [loading, setLoading] = useState(true);

  const currentElement = useSelector(currentElementSelector);

  const currentVehicle = useSelector(selectCurrentVehicle);

  const onOpenDomainModalClick = () => {
    setIsOpenDomainModal(true);
  };

  const onCloseDomainModalClick = () => {
    setIsOpenDomainModal(false);
  };

  const onOpenECUModalClick = () => {
    setIsOpenECUModal(true);
  };

  const onCloseECUModalClick = () => {
    setIsOpenECUModal(false);
  };

  const onOpenHistoryView = () => {
    setIsOpenHistoryView(true);
  };

  const onCloseHistoryView = () => {
    setIsOpenHistoryView(false);
  };

  const onOpenInitiateView = () => {
    setIsOpenInitiateView(true);
  };

  const onOpenRAModal = () => {
    setIsOpenRAModal(true);
  };
  const onCloseRAModal = () => {
    setIsOpenRAModal(false);
  };

  const onApplyModal = () => {
    console.log('apply'); // temp
  };

  const getVehicle = async () => {
    if (!currentVehicle) {
      setVehicle(null);
      return;
    }
    const currentVehicleData = await getCurrentVehicle(currentVehicle);
    setVehicle(currentVehicleData);
  };

  const onCloseInitiateView = () => {
    setIsOpenInitiateView(false);
    getVehicle();
  };

  const getController = async (vehicleId: string) => {
    try {
      setLoading(true);
      const { gateway, type } = await getVehicleGateway(vehicleId);
      setGateway({ gateway, type });
      dispatch(setCurrentElement({ type: EAside[type], id: gateway?.id }));
      setLoading(false);
    } catch (error) {
      dispatch(setCurrentElement(null));
    }
  };

  const getArchitectureSchemeData = async (id: string) => {
    const newSchemeData = await getSchemeData(id);
    setSchemeData(newSchemeData);
    setLoading(false);
  };

  useEffect(() => {
    if (!isOpenDomainModal || isOpenECUModal) {
      getArchitectureSchemeData(currentVehicle);
    }
  }, [currentVehicle, isOpenDomainModal, isOpenECUModal]);

  useEffect(() => {
    getVehicle();
  }, [currentVehicle]);

  useEffect(() => {
    if (currentVehicle && !gateway && !isOpenECUModal) {
      getController(currentVehicle);
    }
  }, [currentVehicle, isOpenECUModal]);

  useEffect(() => {
    if (vehicle) {
      const subtitles = [];
      if (vehicle.model) {
        subtitles.push(vehicle.model);
      }

      if (vehicle.brand?.name) {
        subtitles.push(vehicle?.brand?.name);
      }
      dispatch(setHeader(subtitles));
    }
  }, [vehicle]);

  return (
    <div className="content" id="architecture">
      <ArchitectureControls
        vehicle={vehicle}
        onOpenDomainModalClick={onOpenDomainModalClick}
        onOpenHistoryView={onOpenHistoryView}
      />
      <div className="page-content">
        {loading ? (
          <Preloader />
        ) : (
          <>
            <div className="page-thumb">
              <div className="architecture-scheme" id="architectureScheme">
                <ArchitectureScheme
                  gatewayInfo={gateway}
                  schemeData={schemeData}
                  onECUCreateClick={onOpenECUModalClick}
                />
              </div>
              <div className="architecture-controls">
                <div className="page-controls-main">
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-grid" />
                    <Btn btnStatus="btn-control" btnIcon="icon-list" />
                  </div>
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-maximize" />
                    <div className="btn-box-separator" />
                    <Btn btnStatus="btn-control" btnIcon="icon-minus" />
                  </div>
                </div>
                <Btn btnStatus="btn-control" btnIcon="icon-chevrons-right" />
              </div>
            </div>
            <ArchitectureAside currentElement={currentElement} asideStatus="large">
              <ArchitectureDetailsContent
                onAssign={onOpenRAModal}
                element={currentElement}
                isAssignModalOpen={isOpenRAModal}
                onOpenInitiateView={onOpenInitiateView}
                isInitiateModalOpen={isOpenInitiateView}
              />
            </ArchitectureAside>
          </>
        )}
      </div>
      <DomainDetails
        vehicle={vehicle}
        isOpenDomainModal={isOpenDomainModal}
        onCloseDomainModalClick={onCloseDomainModalClick}
      />
      <ECUDetails vehicle={vehicle} isOpen={isOpenECUModal} onClose={onCloseECUModalClick} />
      <AssignRA element={currentElement} isOpen={isOpenRAModal} onClose={onCloseRAModal} />
      {/* Add content to modals */}
      <ModalCenter title="History" isOpen={isOpenHistoryView} onApply={onApplyModal} onClose={onCloseHistoryView} />
      <InitiateRiskAssesment
        element={currentElement}
        vehicle={vehicle}
        isOpenInitiateView={isOpenInitiateView}
        onCloseInitiateView={onCloseInitiateView}
      />
    </div>
  );
};
