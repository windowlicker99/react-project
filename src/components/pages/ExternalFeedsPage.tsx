import React, { FC, useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import { Heading } from '@components/common/Heading';
import { Table } from '@components/page-contents/Table';
import { Aside } from '@components/common/Aside';
import { CardAlert } from '@components/common/CardAlert';
import { ETableTypes, IAlert } from '@interfaces/interfaces';
import { selectColumnsWithExpandAllDropdownList } from '@store/selectors/columns';
import { getData } from '@utils/getElements';
import { API_PATHS } from '@constants/paths';
import { Preloader } from '@components/common/Preloader';

export const ExternalFeedsPage: FC = () => {
  const [alerts, setAlerts] = useState<IAlert[]>([]);
  const [loading, setLoading] = useState(true);

  const columns = useSelector(selectColumnsWithExpandAllDropdownList);

  const getAlerts = async () => {
    const newAlerts: IAlert[] = await getData(API_PATHS.alerts);
    setAlerts(newAlerts);
  };

  useEffect(() => {
    getAlerts();
  }, []);

  return (
    <Tabs className="page-holder" id="feeds">
      <div className="aside-holder">
        {!loading ? <Heading headingText="Car Alerts" headingNumber={`(${alerts.length})`} /> : null}
        <Aside asideStatus="small transparent">
          <TabList>
            {alerts.map((alert) => (
              <Tab className="tab-item" selectedClassName="active" key={`tab-${alert.name}`}>
                <CardAlert alert={alert} />
              </Tab>
            ))}
          </TabList>
        </Aside>
      </div>

      {alerts.map((alert) => (
        <TabPanel className="tab-content" key={`tabpanel-${alert.name}`}>
          <Heading headingText="Vehicle Types" headingNumber={`(${alert?.vehicleIds.length})`} subtitle={alert.name} />
          <div className="box-table">
            <Table
              changeLoadingState={setLoading}
              tableType={ETableTypes.feeds}
              columns={columns}
              params={{ alertId: alert.id }}
              showHeader
            />
          </div>
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
    </Tabs>
  );
};
