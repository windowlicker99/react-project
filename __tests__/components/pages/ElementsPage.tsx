import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ElementsPage } from '@components/pages/ElementsPage';
import { store } from '@store/store';

describe('ElementsPage tests', () => {
  it('Should render', () => {
    const { container } = render(
      <Provider store={store}>
        <ElementsPage />
      </Provider>
    );
    expect(container.getElementsByClassName('content').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ElementsPage />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
