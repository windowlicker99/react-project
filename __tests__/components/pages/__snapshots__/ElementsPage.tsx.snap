// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsPage tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
    data-tabs="true"
    id="elements"
  >
    <div
      class="box page-controls"
    >
      <div
        class="page-controls-content flex-end"
      >
        <div
          class="page-controls-item"
        >
          <div
            class="mr-15"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label search-label"
                for="exampleFormControlInput1"
              >
                What are you looking for?
              </label>
              <div
                class="search-group"
              >
                <div
                  class="search-icon"
                >
                  <span
                    class="icon-loupe"
                  />
                </div>
                <input
                  class="form-control search-input"
                  id="exampleFormControlInput1"
                  placeholder="Search"
                  type="email"
                  value=""
                />
                <button
                  class="search-btn"
                  type="button"
                >
                  <span
                    class="icon-cross"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="mr-15"
          >
            <button
              class="btn-xs-primary"
              type="button"
            >
              <span
                class="icon-filter-blue"
              />
            </button>
          </div>
        </div>
        <div
          class="page-controls-item"
        >
          <button
            class="btn btn-lg-primary"
            type="button"
          >
            <span
              class="btn-icon icon-columns"
            />
            <span
              class="btn-text"
            >
              Customize Columns
            </span>
          </button>
        </div>
      </div>
      <div
        class="page-filter"
      >
        <div
          class="page-filter-inner"
        />
        <div
          class="page-filter-btn"
        >
          <button
            class="btn btn-xs-secondary mr-15"
            type="button"
          >
            <span
              class="btn-icon icon-close"
            />
            <span
              class="btn-text"
            />
          </button>
          <button
            class="btn btn-lg-outline"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
      <ul
        class="page-tabs"
        role="tablist"
      >
        <li
          aria-controls="react-tabs-9"
          aria-disabled="false"
          aria-selected="true"
          class="page-tabs-item page-tabs-item active"
          id="react-tabs-8"
          role="tab"
          tabindex="0"
        >
          Domains
        </li>
        <li
          aria-controls="react-tabs-11"
          aria-disabled="false"
          aria-selected="false"
          class="page-tabs-item"
          id="react-tabs-10"
          role="tab"
        >
          ECUs
        </li>
        <li
          aria-controls="react-tabs-13"
          aria-disabled="false"
          aria-selected="false"
          class="page-tabs-item"
          id="react-tabs-12"
          role="tab"
        >
          Functions
        </li>
        <li
          aria-controls="react-tabs-15"
          aria-disabled="false"
          aria-selected="false"
          class="page-tabs-item"
          id="react-tabs-14"
          role="tab"
        >
          Data
        </li>
      </ul>
    </div>
    <div
      aria-labelledby="react-tabs-8"
      class="react-tabs__tab-panel box-table elements-content"
      id="react-tabs-9"
      role="tabpanel"
    >
      <table
        class="table"
        role="table"
      >
        <thead
          class="table-header"
        >
          <tr
            class="table-row"
            role="row"
          >
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Model
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Platform
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Name
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Domain Usage
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Communication
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Online Connectivity
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Author
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Last Update
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Color
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  Actions
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="table-body"
          role="rowgroup"
        />
      </table>
    </div>
    <div
      aria-labelledby="react-tabs-10"
      class="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-12"
      class="react-tabs__tab-panel"
      id="react-tabs-13"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-14"
      class="react-tabs__tab-panel"
      id="react-tabs-15"
      role="tabpanel"
    />
    <footer
      class="footer"
    >
      <div
        class="footer-inner "
      >
        <span
          class="footer-text"
        >
          Showing 0 of 0 Domains
        </span>
        <nav
          aria-label="pagination"
          class="pagination"
        >
          <ul
            class="pagination-list"
          >
            <li
              class="page-item disabled"
              role="presentation"
            />
            <li
              class="page-item"
              role="presentation"
            />
          </ul>
        </nav>
      </div>
    </footer>
  </div>
  <div
    class="preloader"
  >
    <div
      class="preloader-circle"
    />
    <span
      class="preloader-label"
    >
      Loading data...
    </span>
  </div>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-heading"
        >
          <div
            class="modal-header-content"
          >
            <span
              class="modal-title"
            >
              Customize columns
            </span>
            <span
              class="modal-subtitle"
            />
          </div>
          <button
            class="btn btn-xs-secondary"
            type="button"
          >
            <span
              class="icon-close"
            />
          </button>
        </div>
      </div>
      <div
        class="modal-body"
      >
        <div
          class="modal-info"
        >
          <form>
            <ul
              class="customise-list"
            >
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="model"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="model"
                      name="model"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Model
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="platform"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="platform"
                      name="platform"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Platform
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="name"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="name"
                      name="name"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Name
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="usage"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="usage"
                      name="usage"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Domain Usage
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="communication"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="communication"
                      name="communication"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Communication
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="connectivity"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="connectivity"
                      name="connectivity"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Online Connectivity
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="author"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="author"
                      name="author"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Author
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="update"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="update"
                      name="update"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Last Update
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="color"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="color"
                      name="color"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Color
                    </span>
                  </label>
                </div>
              </li>
              <li
                class="customise-item"
              >
                <button
                  class="customise-move"
                  type="button"
                />
                <div
                  class="checkbox"
                >
                  <label
                    class="checkbox-label"
                    for="actions"
                  >
                    <input
                      checked=""
                      class="checkbox-input"
                      id="actions"
                      name="actions"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-mark"
                    />
                    <span
                      class="checkbox-title"
                    >
                      Actions
                    </span>
                  </label>
                </div>
              </li>
            </ul>
          </form>
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <div
          class="btn-group-stretch"
        >
          <button
            class="btn btn-lg-outline mr-15"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Cancel
            </span>
          </button>
          <button
            class="btn btn-lg-primary"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
